src/WinGet.RestSource.Infrastructure/Templates/AppConfig/appconfig.json

```diff
@@ -0,0 +1,61 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "appConfigName": {
+            "type": "string",
+            "defaultValue": "[concat('appconfig', uniquestring(resourceGroup().id))]",
+            "metadata": {
+                "description": "Name of the Azure App Config"
+            }
+        },
+        "location": {
+            "type": "string",
+            "defaultValue": "[resourceGroup().location]",
+            "metadata": {
+                "description": "App Config Location."
+            }
+        },
+        "genevaLogging": {
+            "type": "bool",
+            "defaultValue": false,
+            "metadata": {
+                "description": "Feature flag to use Geneva Monitoring."
+            }
+        }
+    },
+    "variables": {
+        "genevaLoggingValue": {
+            "id": "GenevaLogging",
+            "description": "Feature flag to use Geneva Monitoring.",
+            "enabled": "[parameters('genevaLogging')]",
+            "conditions": { "client_filters": [] }
+        }
+    },
+    "resources": [
+        {
+            "name": "[parameters('appConfigName')]",
+            "type": "Microsoft.AppConfiguration/configurationStores",
+            "apiVersion": "2020-07-01-preview",
+            "location": "[parameters('location')]",
+            "sku": {
+                "name": "standard"
+            },
+            "properties": {},
+            "resources": [
+                {
+                    "name": "[concat('.appconfig.featureflag~2F', variables('genevaLoggingValue').id)]",
+                    "type": "keyValues",
+                    "apiVersion": "2020-07-01-preview",
+                    "dependsOn": [
+                        "[parameters('appConfigName')]"
+                    ],
+                    "properties": {
+                        "value": "[string(variables('genevaLoggingValue'))]",
+                        "contentType": "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+                    }
+                }
```

Comment:
A year and a half a go when I made this for wingetsvc, there was not even documentation on how to produce ARM templates for AppConfig, specifically for feature flags. So I just had to figure it out and do this

Not that is going to happen, but I always find it tedious to add new feature flags. You need to add the parameter, add the variable, add the resource which is pretty much copy paste, find and replace and hope you didn't mess up. Then add the value per each region per environment.

I want it to make it such as you just define one json file per env some script will produce the parameter templates for the different regions at release time. that way you just change once per env not n per region in env.

And then also have a pseudo template appconfig.json that only takes the name of the feature flags and produce this file at release time too. Or even just as build artifact.

I think it will make changes way easier. I do have a bug (for a year now) but I never got time to do fix it. Maybe one day, right? ... right?

Reply:
We can probably use a copy for this. It'll take a little more thought then we have time for right this second, but lets sync on this.